resources/aws_ec2.go

```diff
@@ -677,3 +682,29 @@ func (obj *AwsEc2Res) UnmarshalYAML(unmarshal func(interface{}) error) error {
 func (obj *AwsEc2Res) prependName() string {
 	return AwsPrefix + obj.GetName()
 }
+
+// snsServer returns an http server used to listen for sns messages.
+func (obj *AwsEc2Res) snsServer() *http.Server {
+	addr := obj.WatchListenAddr
+	// if addr is a port
+	if _, err := strconv.Atoi(obj.WatchListenAddr); err == nil {
+		addr = fmt.Sprintf(":%s", obj.WatchListenAddr)
+	}
+	handler := http.HandlerFunc(obj.snsPostHandler)
+	return &http.Server{
+		Addr:    addr,
+		Handler: handler,
+	}
+}
+
+// snsPostHandler listens for posts on the SNS Endpoint.
+func (obj *AwsEc2Res) snsPostHandler(w http.ResponseWriter, req *http.Request) {
+	if req.Method != "POST" {
+		http.Error(w, "Invalid request method", http.StatusMethodNotAllowed)
+		return
+	}
+	post, _ := ioutil.ReadAll(req.Body)
```

Comment:
why don't we check this error? (sorry)
more importantly, what do we even ever do in this method? it seems basically useless... (or is something coming later?)

Reply:
Yeah. This is just so it has something to do. Once snsSubscribe is done, this will use json.Decoder to read the post, and we can get rid of the ioutil stuff.